hibernate-core/src/main/java/org/hibernate/service/spi/ServiceBinding.java

```diff
@@ -46,28 +52,45 @@ public ServiceBinding(ServiceLifecycleOwner lifecycleOwner, ServiceInitiator<R>
 		this.serviceInitiator = serviceInitiator;
 	}
 
-	public ServiceLifecycleOwner getLifecycleOwner() {
-		return lifecycleOwner;
-	}
-
 	public Class<R> getServiceRole() {
 		return serviceRole;
 	}
 
-	public ServiceInitiator<R> getServiceInitiator() {
-		return serviceInitiator;
-	}
+	public synchronized R getService() {
+		if ( service == null ) {
```

Comment:
Since service is `volatile`, we can move the main `if` clause in a `synchronized(this)` block so that we don't have to acquire a lock after the service is initialized.